.github/workflows/build-pr.yml

```
@@ -0,0 +1,82 @@
+name: build test site on trigger comment
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: khan/pull-request-comment-trigger@master
+        id: check
+        with:
+          trigger: '#deploy'
+          reaction: rocket
+        env:
+          GITHUB_TOKEN: '${{ secrets.GITHUB_TOKEN }}'
+
+      # this step is needed to get the PR ref from an issue comment
+      # https://github.com/actions/checkout/issues/331
+      - uses: actions/github-script@v3
+        if: steps.check.outputs.triggered == 'true'
+        id: get-pr
+        with:
+          script: |
+            const request = {
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.issue.number
+            }
+            core.info(`Getting PR #${request.pull_number} from ${request.owner}/${request.repo}`)
+            try {
+              const result = await github.pulls.get(request)
+              return result.data
+            } catch (err) {
+              core.setFailed(`Request failed with error ${err}`)
+            }
+      - uses: actions/checkout@v2
+        if: steps.check.outputs.triggered == 'true'
+        with:
+          repository: ${{ fromJSON(steps.get-pr.outputs.result).head.repo.full_name }}
+          ref: ${{ fromJSON(steps.get-pr.outputs.result).head.sha }} # or .head.ref for branch name
+
+      - name: Checkout mathlib
+        if: steps.check.outputs.triggered == 'true'
+        run: git clone https://github.com/leanprover-community/mathlib
+
+      - name: install elan
+        if: steps.check.outputs.triggered == 'true'
+        run: |
+          set -o pipefail
+          curl https://raw.githubusercontent.com/Kha/elan/master/elan-init.sh -sSf | sh -s -- --default-toolchain none -y
+          cd mathlib
+          ~/.elan/bin/lean --version
+          echo "$HOME/.elan/bin" >> $GITHUB_PATH
+
+      - name: install Python
+        if: steps.check.outputs.triggered == 'true'
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+
+      - name: install Python dependencies
+        if: steps.check.outputs.triggered == 'true'
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: run leanproject
+        if: steps.check.outputs.triggered == 'true'
+        run: |
+          cd mathlib
+          leanproject up
+
+      - name: generate docs
+        if: steps.check.outputs.triggered == 'true'
+        run: ./deploy_docs.sh "mathlib" ".." "mathlib" "leanprover-community" "mathlib_docs_demo" "true"
```

```suggestion
        run: |
          wget https://raw.githubusercontent.com/leanprover-community/doc-gen/master/deploy_docs.sh
          ./deploy_docs.sh "mathlib" ".." "mathlib" "leanprover-community" "mathlib_docs_demo" "true"
```

@gebner this would solve the security problem you're worried about, right? I think it's reasonable to ask that we not use `#deploy` to test the deployment script itself.